Amc data questionnaire updates

dodwho opened the pull request:
### :pushpin: References

-   **Issue:** Closes #?

### :memo: Implementation

### :video_camera: Screenshots/Screen capture

### :fire: Testing

anagperal commented on src/webapp/components/current-data-submission/amc-questionnaires/mappers/generalAMCQuestionnaireMapper.ts:
```diff
@@ -39,7 +39,7 @@ export function mapFormStateToGeneralAMCQuestionnaire(
     const detailOnShortageInPrivateSector = getStringFieldValue("detailOnShortageInPrivateSector", allFields);
     const generalComments = getStringFieldValue("generalComments", allFields);
-    const antibacterials = yesNoOption.getValueFromBoolean(getBooleanFieldValue("antibacterials", allFields));
+    const antibiotics = yesNoOption.getValueFromBoolean(getBooleanFieldValue("antibacterials", allFields));
```
Since the ID of the field has change in `generalAMCQuestionnaireFieldIds` from antibacterials to antibiotics, it needs to be changed in `getBooleanFieldValue` as well:  `const antibiotics = yesNoOption.getValueFromBoolean(getBooleanFieldValue("antibiotics", allFields));`

anagperal commented on src/webapp/components/form/widgets/NumberInput.tsx:
```diff
@@ -0,0 +1,93 @@
+                    onChange={e => {
+                        const val = e.target.value;
+                        // Prevent empty string — fallback to 0
+                        setInputValue(val === "" ? "0" : val);
```
This generates somewhat weird behavior when you try to delete all the field

[Screencast from 2025-08-07 09-00-01.webm](https://github.com/user-attachments/assets/8837ff18-dcdf-4417-b6dc-f74ae5ea943c)

Perhaps you could increase the bounce delay to 500,  and add this if at the beginning of the useEffect:

```
    useEffect(() => {
            if (debouncedValue === "") {
                setInputValue("0");
                onChange(0);
            }

            const parsed = Number(debouncedValue);
            if (!isNaN(parsed)) {
                onChange(parsed);
            }
        }, [debouncedValue, onChange])

``` 

and in the onChange directly set the value setInputValue(e.target.value);

[Screencast from 2025-08-07 08-58-45.webm](https://github.com/user-attachments/assets/c7962285-c96c-4fdb-b721-bba0798a96fb)

anagperal reviewed (changes requested):
Thanks @dodwho ! I've left you just a couple of comments.

anagperal commented on .gitattributes:
```diff
@@ -0,0 +1 @@
+* text=auto eol=lf
```
This file is not required in the PR, would be best to remove the file or add it to .gitignore file

anagperal reviewed (changes requested):
@dodwho thanks! Everything seems to be ok, only a minor comment about .gitattributes file.Also The option set for AMR_GLASS_AMC_DE_VOL_COMM and AMR_GLASS_AMC_DE_VOL_HOSP data elements needs to be changed to the one that includes the option ‘unknown’

dodwho commented:
@anagperal Thanks for this. I've taken your point and removed .gitattributes from being tracked by git. 
Also great point about the data elements still having the older optionSet. I've updated this in DHIS2 now.

anagperal reviewed (approved):
thanks @dodwho !! all good!
